src/views/StoryViewer.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <div class="viewer" :style="currentFrame.style">
+    <div class="progress-container">
+      <div v-for="n in storyLength" :key="n" class="progress">
+        <div v-if="currentFrame.id === n" class="loading"></div>
+        <div v-else-if="currentFrame.id > n" class="loaded"></div>
+      </div>
+    </div>
+    <div class="story-text" v-if="currentFrame.text">{{ currentFrame.text }}</div>
+    <div class="restart-btn" v-if="storyEnd" @click="restartStory">{{ $i18n('btn-restart-story') }}</div>
+  </div>
+</template>
+
+<script>
+import { mapGetters, mapActions } from 'vuex'
+export default {
+  name: 'StoryViewer',
+  data: () => {
+    return {
+      frameDuration: 2000,
+      storyEnd: false
+    }
+  },
+  computed: mapGetters(['currentFrame', 'storyLength']),
+  methods: {
+    ...mapActions(['selectFrame']),
+    playNextFrame: function() {
+      const timeoutId = setTimeout( () => {
+        this.selectFrame(this.currentFrame.id + 1)
+        clearTimeout(timeoutId)
+      }, this.frameDuration)
+    },
+    restartStory: function() {
+      this.storyEnd = false
+      this.selectFrame(1)
+    },
+    endStory: function() {
+      const timeoutId = setTimeout( ()=> {
+        this.storyEnd = true
+        clearTimeout(timeoutId)
+      }, this.frameDuration)
+    }
+  },
+  beforeMount: function() {
+    if (this.currentFrame.id > 1) {
+      this.restartStory()
+    }
+  },
+  mounted: function() {
+    if (this.currentFrame.id < this.storyLength) {
+      this.playNextFrame()
+    }
+  },
+  updated: function() {
+    if (this.currentFrame.id < this.storyLength) {
+      this.playNextFrame()
+    } else if(!this.storyEnd) {
+      this.endStory()
+    }
+  }
+}
+</script>
+
+<style scoped>
+  .viewer {
+    height: 100%;
+    width: 100vw;
+    position: relative;
+    text-align: center;
+  }
+  .story-text {
+    position: absolute;
+    bottom: 80px;
+    left: 20px;
+    right: 20px;
+    border-radius: 10px;
+    background-color: white;
+    margin: 0;
+    padding: 10px;
+  }
+  .restart-btn {
```

Comment:
one more thing, not really necessary for mobile but nice to have in desktop
having the `cursor: pointer` to make it looks like button when hover.

Reply:
I'm going to add the cursor and merge it. Eventually we need to all use a single button component.